noxfile.py

```diff
@@ -87,6 +87,8 @@ def test(session):
 
     # Build source distribution
     sdist_dir = os.path.join(session.virtualenv.location, "sdist")
+    if os.path.exists(sdist_dir):
```

Comment:
Should we try to match the actual way we release i.e. build sdist with latest available 3.x and then test with all/selected?

Reply:
I don't think so? Like, it'd get too complex too quickly since I don't think these tools let you invoke multiple versions of Python for the same job (nox, tox).

Anyway, I want to move the sdist generation into a function and switch to editable installs unless on CI, since they're cheaper and faster.

Reply:
OK, I assumed (incorrectly, after checking the docs) that nox would have a dependency mechanism so we could e.g. have this session depend on another one that used a different Python for its venv.